rules_python/piptool.py

```diff
@@ -93,6 +48,47 @@ def repository_name(self):
 parser.add_argument('--directory', action='store',
                     help=('The directory into which to put .whl files.'))
 
+def determine_possible_extras(whls):
+  whl_map = {
+    whl.distribution(): whl
+    for whl in whls
+  }
+  def parse_requirement(name):
+    if '[' not in name:
```

Comment:
Can you use `pkg_resources.Requirement.parse(name)` instead?  This might be a little brittle.

Reply:
Done, thanks for the pointer.